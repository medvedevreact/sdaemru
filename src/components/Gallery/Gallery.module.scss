.categoryCarts {
  display: flex;
  align-items: center;

  list-style-type: none;

  padding-left: 0;
  gap: 60px;
  flex-wrap: wrap;
  @media (max-width: 1060px) {
    gap: 40px;
  }
  @media (max-width: 860px) {
    gap: 60px;
    justify-content: center;
  }
  @media (max-width: 700px) {
    gap: 30px;
  }

  li:first-child {
    width: 800px;
    height: 400px;
    @media (max-width: 1280px) {
      width: 665px;
      height: 332px;
    }
    @media (max-width: 1060px) {
      width: 500px;
      height: 250px;
    }
    @media (max-width: 860px) {
      width: 665px;
      height: 332px;
    }
    @media (max-width: 700px) {
      width: 85%;
      height: 200px;
    }
  }
}

.categoryCart {
  width: 370px;
  height: 260px;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  border-radius: 15px;
  position: relative;
  margin-left: 0;
  @media (max-width: 1280px) {
    width: 300px;
    height: 200px;
  }
  @media (max-width: 1060px) {
    width: 230px;
    height: 150px;
  }
  @media (max-width: 860px) {
    width: 300px;
    height: 200px;
  }
  @media (max-width: 700px) {
    width: 85%;
    min-height: 120px;
  }
}
.title {
  position: absolute;
  top: 15px;
  left: 15px;
  color: rgb(115, 79, 222);
  box-shadow: 1px 1px 3px rgba(0, 0, 0, 0.2);
  @media (max-width: 1060px) {
    font-size: 15px;
  }
}
.name {
  position: absolute;
  top: 50px;
  left: 15px;
  color: white;
  box-shadow: 1px 1px 3px rgba(0, 0, 0, 0.2);
  @media (max-width: 1060px) {
    font-size: 15px;
    top: 40px;
  }
}
.icon {
  position: absolute;
  cursor: pointer;
  bottom: 7px;
  right: 7px;
}

.wrapper {
  margin-bottom: 60px;
  width: calc(100% - 300px);
  @media (max-width: 860px) {
    width: 100%;
  }
}

.svgBorder {
  width: 30px;
  height: 30px;
  border-radius: 50%;
  padding: 12px;
  border: 1px solid white;
  position: absolute;
  cursor: pointer;
  bottom: 20px;
  right: 20px;
}

.section {
  display: flex;
  @media (max-width: 860px) {
    justify-content: center;
  }
}

aside {
  @media (max-width: 860px) {
    display: none;
  }
}
.asideList {
  list-style-type: none;
  margin-bottom: 30px;
  display: flex;
  flex-direction: column;
  gap: 7px;
  .asideLi {
    color: gold;
    cursor: pointer;
  }
}
.asideTitle {
  margin-bottom: 10px;
  cursor: pointer;
}
