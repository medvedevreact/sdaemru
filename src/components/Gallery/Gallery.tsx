import React from "react";
import styles from "./Gallery.module.scss";

import { useNavigate } from "react-router-dom";

const categories = [
  {
    title: "Снять квартиру",
    name: "Квартиры",
    photo: "src/img/appartment.jpg",
    link: "appartments",
  },
  {
    title: "Снять Коттедж",
    name: "Коттеджи",
    photo: "src/img/house.png",
    link: "houses",
  },
  {
    title: "Арендовать авто",
    name: "Авто",
    photo: "src/img/car.jpg",
    link: "auto",
  },
];

export const Gallery = () => {
  const navigate = useNavigate();
  return (
    <div className={styles.section}>
      <div className={styles.wrapper}>
        <ul className={styles.categoryCarts}>
          {categories.map((el) => (
            <li
              className={styles.categoryCart}
              key={el.name}
              style={{ backgroundImage: `url(${el.photo})` }}
            >
              <div className={styles.content}>
                <h3 className={styles.title}>{el.title}</h3>
                <h2 className={styles.name}>{el.name}</h2>
                <div
                  className={styles.svgBorder}
                  onClick={() => {
                    navigate(`/listings/${el.link}`);
                  }}
                >
                  <svg
                    width="15px"
                    height="15px"
                    viewBox="0 0 1024 1024"
                    className={styles.icon}
                    version="1.1"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 120.768L306.432 64 768 512l-461.568 448L256 903.232 659.072 512z"
                      fill="#ffffff"
                    />
                  </svg>
                </div>
              </div>
            </li>
          ))}
        </ul>
      </div>
      <aside>
        <div>
          <h2 className={styles.asideTitle}> Квартиры</h2>
          <ul className={styles.asideList}>
            <li className={styles.asideLi}>Однокомнатные квартиры</li>
            <li className={styles.asideLi}>Двухкомнатные квартиры</li>
            <li className={styles.asideLi}>Трёхкомнатные квартиры</li>
          </ul>
        </div>
        <div>
          <h2 className={styles.asideTitle}>Коттеджи</h2>
          <ul className={styles.asideList}>
            <li className={styles.asideLi}>Малые коттеджи</li>
            <li className={styles.asideLi}>Средние коттеджи</li>
            <li className={styles.asideLi}>Большие коттеджи</li>
          </ul>
        </div>
      </aside>
    </div>
  );
};
