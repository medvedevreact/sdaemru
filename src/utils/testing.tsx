import { MemoryRouter, MemoryRouterProps } from "react-router-dom";
import { render, RenderOptions } from "@testing-library/react";
import { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { store } from "../store";

interface RenderWithRouterOptions
  extends MemoryRouterProps,
    Omit<RenderOptions, "wrapper"> {}

export const renderWithRouter = (
  component: ReactElement,
  options: RenderWithRouterOptions = {}
) => {
  const Wrapper = ({ children }: { children?: ReactNode }) => (
    <MemoryRouter {...options}>
      <Provider store={store}>{children}</Provider>
    </MemoryRouter>
  );

  return render(component, { wrapper: Wrapper, ...options });
};
