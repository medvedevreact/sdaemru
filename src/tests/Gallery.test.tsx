import "@testing-library/jest-dom";
import { renderWithRouter } from "../utils/testing";
import { screen } from "@testing-library/react";
import { Gallery } from "../components/Gallery/Gallery";

describe("desc component", () => {
  test("render", () => {
    renderWithRouter(<Gallery />);
    expect(screen.getByRole("gallery")).toMatchSnapshot();
  });
});
