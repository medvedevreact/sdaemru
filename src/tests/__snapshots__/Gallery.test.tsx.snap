// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`desc component render 1`] = `
<div
  class="section"
  role="gallery"
>
  <div
    class="wrapper"
  >
    <ul
      class="categoryCarts"
    >
      <li
        class="categoryCart"
        role="img"
        style="background-image: url(img/appartment.jpg);"
      >
        <div
          class="content"
        >
          <h3
            class="title"
          >
            Снять квартиру
          </h3>
          <h2
            class="name"
          >
            Квартиры
          </h2>
          <div
            class="svgBorder"
          >
            <svg
              class="icon"
              height="15px"
              version="1.1"
              viewBox="0 0 1024 1024"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 120.768L306.432 64 768 512l-461.568 448L256 903.232 659.072 512z"
                fill="#ffffff"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="categoryCart"
        role="img"
        style="background-image: url(img/house.png);"
      >
        <div
          class="content"
        >
          <h3
            class="title"
          >
            Снять Коттедж
          </h3>
          <h2
            class="name"
          >
            Коттеджи
          </h2>
          <div
            class="svgBorder"
          >
            <svg
              class="icon"
              height="15px"
              version="1.1"
              viewBox="0 0 1024 1024"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 120.768L306.432 64 768 512l-461.568 448L256 903.232 659.072 512z"
                fill="#ffffff"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="categoryCart"
        role="img"
        style="background-image: url(img/car.jpg);"
      >
        <div
          class="content"
        >
          <h3
            class="title"
          >
            Арендовать авто
          </h3>
          <h2
            class="name"
          >
            Авто
          </h2>
          <div
            class="svgBorder"
          >
            <svg
              class="icon"
              height="15px"
              version="1.1"
              viewBox="0 0 1024 1024"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 120.768L306.432 64 768 512l-461.568 448L256 903.232 659.072 512z"
                fill="#ffffff"
              />
            </svg>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <aside>
    <div>
      <h2
        class="asideTitle"
      >
         Квартиры
      </h2>
      <ul
        class="asideList"
      >
        <li
          class="asideLi"
        >
          Однокомнатные квартиры
        </li>
        <li
          class="asideLi"
        >
          Двухкомнатные квартиры
        </li>
        <li
          class="asideLi"
        >
          Трёхкомнатные квартиры
        </li>
      </ul>
    </div>
    <div>
      <h2
        class="asideTitle"
      >
        Коттеджи
      </h2>
      <ul
        class="asideList"
      >
        <li
          class="asideLi"
        >
          Малые коттеджи
        </li>
        <li
          class="asideLi"
        >
          Средние коттеджи
        </li>
        <li
          class="asideLi"
        >
          Большие коттеджи
        </li>
      </ul>
    </div>
  </aside>
</div>
`;
